src/app.rs

```diff
@@ -1886,7 +1865,7 @@ impl<T: Frontend> App<T> {
             Some(tagged_editor)
                 if Some(tagged_editor.clone()) != self.current_component().borrow().path() =>
             {
-                self.open_file(&tagged_editor, OpenFileOption::Focus)?;
+                self.open_file(&tagged_editor, BufferOwner::User, true)?;
```

Comment:
Same here, we should create a method called `open_buffer` that only takes one parameter which is `CanonicalizedPath`, and `open_file` should call `open_buffer` if the requested path is already opened.